source/posix/condition_variable.c

```diff
@@ -92,13 +92,13 @@ int aws_condition_variable_wait_for(
     if (aws_sys_clock_get_ticks(&current_sys_time)) {
         return AWS_OP_ERR;
     }
-
-    time_to_wait += current_sys_time;
+    uint64_t time_to_wait_unsigned = (uint64_t)time_to_wait;
+    time_to_wait_unsigned += current_sys_time;
 
     struct timespec ts;
     uint64_t remainder = 0;
     ts.tv_sec =
-        (time_t)aws_timestamp_convert((uint64_t)time_to_wait, AWS_TIMESTAMP_NANOS, AWS_TIMESTAMP_SECS, &remainder);
+        (time_t)aws_timestamp_convert(time_to_wait_unsigned, AWS_TIMESTAMP_NANOS, AWS_TIMESTAMP_SECS, &remainder);
```

Comment:
nit: why do we need a temp variable? cant we just have time_to_wait + current_sys_time inline?